src/backend/models/lcmdiffusion_setting.py

```diff
@@ -26,6 +27,13 @@ class Lora(BaseModel):
     enabled: bool = False
 
 
+class ControlNetSetting(BaseModel):
+    path: Optional[str] = None  # ControlNet adapter path
+    weight: float = 0.5
+    enabled: bool = False
+    _image: Image = None        # Control image, PIL image
```

Comment:
The underscore tells _pydantic_ to ignore that field when saving or _serializing_ the settings. Otherwise, _pydantic_ complains about that field. Alternatively, instead of a PIL image it could be an image path stored as a simple _str_.